impress-org/give-google-analytics
# Feature: implement google analytics 4

Resolves https://github.com/impress-org/give-google-analytics/issues/56

## Description


This pull request implements Google Analytics 4 API  for tracking donations. Universal Analytics is still supported.

## Visuals
**Settings in WP Dashboard**
<img width="1032" alt="Screenshot 2022-08-05 at 11 25 05 PM" src="https://user-images.githubusercontent.com/1784821/183133758-cb6515c4-f4ff-48ca-a4cd-dc281c50e78d.png">

**Google Analytics dashboard**
<img width="1692" alt="image" src="https://user-images.githubusercontent.com/1784821/183133814-a78181a7-ebaa-4ece-a26f-8784b22d3066.png">


## Testing Instructions


The addon should record `view_item`, `begin_chekcout`, `purchase`, and `refund` google analytics event.

Note:
- Setup google analytics before start testing: https://support.google.com/analytics/answer/9304153?hl=en
- I tested the add-on only with Google Analytics for WordPress by MonsterInsights WordPress plugin.
- Set Google tracking mode in the add-on setting before starting testing.

## Pre-review Checklist
